src/variable.cpp

```diff
@@ -77,6 +90,37 @@ void Variable::swap(Variable &rhs)
     std::swap(this->mPimpl, rhs.mPimpl);
 }
 
+void Variable::addEquivalence(const VariablePtr &variable1, const VariablePtr &variable2)
+{
+    variable1->setEquivalentTo(variable2);
+    variable2->setEquivalentTo(variable1);
+}
+
+VariablePtr Variable::getEquivalentVariable(size_t index)
+{
+    VariableWeakPtr weakEquivalentVariable = mPimpl->mEquivalentVariables.at(index);
+    return weakEquivalentVariable.lock();
+}
+
+size_t Variable::equivalentVariableCount() const
+{
+    return mPimpl->mEquivalentVariables.size();
+}
+
+bool Variable::hasEquivalentVariable(const VariablePtr &equivalentVariable)
+{
+    auto result = mPimpl->findEquivalentVariable(equivalentVariable);
+    return result != mPimpl->mEquivalentVariables.end();
```

Comment:
Line width issue? I think originally this was a hangover from debugging a problem.